### PR TITLE
fix(rust): fix clippy warnings.

### PR DESCRIPTION
**Pull request by twittner**
## Current Behaviour

Upgrading rustc to 1.60 produces some warnings from clippy.

## Proposed Changes

Fix clippy warnings introduced with rustc 1.60

## Checks

- [x] All commits in this Pull Request are [signed](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits).
- [x] All commits in this Pull Request follow the Ockam [commit message convention](https://www.ockam.io/learn/how-to-guides/contributing/CONTRIBUTING#commit-messages).
- [x] I accept the Ockam Community [Code of Conduct](https://www.ockam.io/learn/how-to-guides/high-performance-team/conduct/).
- [x] I have accepted the Ockam [Contributor Licence Agreement](https://www.ockam.io/learn/how-to-guides/contributing/cla/) by adding my Git/Github details in a row at the end of the [CONTRIBUTORS.csv](https://github.com/ockam-network/contributors/blob/master/CONTRIBUTORS.csv) file in a separate pull request to the [ockam-network/contributors](https://github.com/ockam-network/contributors) repository.

### CODE REVIEW

File: implementations/rust/ockam/ockam_core/src/access_control.rs [hunk 1]
```diff
@@ -26,15 +26,15 @@ use crate::{LocalMessage, Result};
 #[async_trait]
 pub trait AccessControl: Send + Sync + 'static {
     /// Return true if the message is allowed to pass, and false if not.
-    async fn is_authorized(&mut self, local_msg: &LocalMessage) -> Result<bool>;
+    async fn is_authorized(&self, local_msg: &LocalMessage) -> Result<bool>;
```
Comment (twittner) [inline implementations/rust/ockam/ockam_core/src/access_control.rs:29]: This is debatable of course. Currently no impl requires unique access, but if the design is intended to do so, we may instead want to add `#[allow(clippy::wrong-self-convention)]`.

**Review by thomcc — APPROVED** [review]

File: implementations/rust/ockam/ockam_core/src/access_control.rs (hunk 1 above)
Comment (thomcc) [reply implementations/rust/ockam/ockam_core/src/access_control.rs:29]: I think there used to be the perception in ockam's code that we should prefer `&mut self`, but I'm strongly in favor of not doing so, and places where we don't need it should remove it. Clippy is right here.

Anybody who needs mutability in this trait can hold onto a Mutex or something.

**Comment by thomcc** [comment]
CC @antoinevg this touches ockam_transport_tcp, but your rebase hasn't happened yet (AFAICT), so I'm going to merge this regardless.

**Comment by thomcc** [comment]
I mean, I'll merge it right away after the rebase. So feel free.

(FWIW I usually just squash stuff like that into a single commit so that rebasing doesnt end up needing to fix code that will get changed/moved/removed as part of a later patch in the rebase. I get not wanting to collapse work and loose history though, and there are other options too -- https://git-scm.com/docs/git-rerere sometimes helps as well)

**Pull request merged**